src/sources/file/mod.rs

```diff
@@ -195,11 +258,18 @@ pub fn file_source(
         let (tx, rx) = futures::sync::mpsc::channel(100);
 
         let messages: Box<dyn Stream<Item = (Bytes, String), Error = ()> + Send> =
-            if let Some(msi) = message_start_indicator {
+            if let Some(ref multiline_config) = multiline_config {
+                Box::new(LineAgg::new(
+                    rx,
+                    multiline_config.try_into().unwrap(), // validated in build
```

Comment:
One last thought. Is there any way for this created `line_agg::Config` to be stored where it is parsed in `build` and used here instead of re-parsing?

Reply:
I tried going that route - but ended up wanting to redo the whole `file` source configuration approach. I think we need to address this systematically - validate and update all the units. In short, the idea I had is to have two layers - a serializable/deserializeable config and a "processed" config, that's been validated and has values converted into the proper types. 
So, in short - I see how ugly the current approach is - and we definitely should fix this, but even if we'll do it just with the `file` source - this is one of the part where I deliberately decided to leave the current approach unchanged.

Reply:
I created https://github.com/timberio/vector/issues/1895